components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt

```diff
@@ -118,4 +152,48 @@ internal object DownloadNotification {
 
         return NOTIFICATION_CHANNEL_ID
     }
+
+    private fun getPauseAction(context: Context, downloadStateId: Long): NotificationCompat.Action {
+        val pauseIntent = createPendingIntent(context, ACTION_PAUSE, downloadStateId)
+
+        // Tell us which download id to pause
+        return NotificationCompat.Action.Builder(
+            0,
+            context.getString(R.string.mozac_feature_downloads_button_pause),
+            pauseIntent
+        ).build()
+    }
+
+    private fun getResumeAction(context: Context, downloadStateId: Long): NotificationCompat.Action {
+        val resumeIntent = createPendingIntent(context, ACTION_RESUME, downloadStateId)
+
+        return NotificationCompat.Action.Builder(
+            0,
+            context.getString(R.string.mozac_feature_downloads_button_resume),
+            resumeIntent
+        ).build()
+    }
+
+    private fun getCancelAction(context: Context, downloadStateId: Long): NotificationCompat.Action {
+        val cancelIntent = createPendingIntent(context, ACTION_CANCEL, downloadStateId)
+
+        return NotificationCompat.Action.Builder(
+            0,
+            context.getString(R.string.mozac_feature_downloads_button_cancel),
+            cancelIntent
+        ).build()
+    }
+
+    private fun createPendingIntent(context: Context, action: String, downloadStateId: Long): PendingIntent {
+        val intent = Intent(action)
+        intent.setPackage(context.applicationContext.packageName)
+
+        val bundleExtra = Bundle()
+        bundleExtra.putLong(EXTRA_DOWNLOAD_ID, downloadStateId)
+        intent.putExtras(bundleExtra)
```

Comment:
nit: Use `putLongExtra` for a one-liner
```suggestion
        intent.putLongExtra(EXTRA_DOWNLOAD_ID, downloadStateId)
```

Reply:
Done!